### PR TITLE
ZK-3930: Provide a camera component

### PR DESCRIPTION
**Pull request by JiaChui**

### CODE REVIEW

File: zcommon/src/org/zkoss/video/AVideo.java [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+	 * If the stream shall be created each time {@link #getStreamData} is called,
+	 * you can pass {@link #DYNAMIC_STREAM} to the data argument, and then
+	 * override {@link #getStreamData}.
+	 * <p>
+	 * <p>Note: the caller of {@link #getStreamData} has to close
+	 * the returned input stream.
+	 */
+	public AVideo(String name, InputStream isdata) throws IOException {
+		if (isdata == null)
+			throw new IllegalArgumentException("null stream");
```
Comment (DevChu) [inline zcommon/src/org/zkoss/video/AVideo.java]: how about "stream cannot be null"? that would be a completed sentence

File: zcommon/src/org/zkoss/video/AVideo.java [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+		_data = null;
+		_file = null;
+	}
+	
+	/**
+	 * Constructs a video with a file.
+	 */
+	public AVideo(File file) {
+		if (file == null)
+			throw new IllegalArgumentException("null url");
```
Comment (DevChu) [inline zcommon/src/org/zkoss/video/AVideo.java]: It is file, not url.
how about "file cannot be null"? that would be a completed sentence

File: zcommon/src/org/zkoss/video/AVideo.java [hunk 3]
```diff
@@ -0,0 +1,259 @@
...
+		_url = null;
+		_file = null;
+	}
+	
+	/**
+	 * Constructs a video with an URL.
+	 */
+	public AVideo(URL url) {
+		if (url == null)
+			throw new IllegalArgumentException("null url");
```
Comment (DevChu) [inline zcommon/src/org/zkoss/video/AVideo.java]: how about "url cannot be null"? that would be a completed sentence

File: zcommon/src/org/zkoss/video/AVideo.java [hunk 4]
```diff
@@ -0,0 +1,259 @@
...
+	 * the returned input stream.
+	 */
+	public AVideo(InputStream is) throws IOException {
+		this(null, is);
+	}
+	
+	private String getName(URL url) {
+		String name = url.getPath();
+		if (name != null) {
+			{
```
Comment (DevChu) [inline zcommon/src/org/zkoss/video/AVideo.java]: extra { ?

File: zcommon/src/org/zkoss/video/AVideo.java [hunk 5]
```diff
@@ -0,0 +1,259 @@
...
+ * <p>AVideo is serializable, but, if you are using InputStream,
+ * you have to extend this class, and provide the implementation to
+ * serialize and deserialize {@link #_isdata}. (Since 8.5.2)
+ */
+public class AVideo implements Video, java.io.Serializable {
+	/**
+	 * Used if you want to implement a media whose input stream is created
+	 * dynamically each time {@link #getStreamData} is called.
+	 *
+	 * @see #AVideo(String, , InputStream)
```
Comment (DevChu) [inline zcommon/src/org/zkoss/video/AVideo.java]: two comma?

File: zcommon/src/org/zkoss/video/AVideo.java (hunk 1 above)
Comment (JiaChui) [reply zcommon/src/org/zkoss/video/AVideo.java]: ok

File: zcommon/src/org/zkoss/video/AVideo.java (hunk 2 above)
Comment (JiaChui) [reply zcommon/src/org/zkoss/video/AVideo.java]: ok

File: zcommon/src/org/zkoss/video/AVideo.java (hunk 3 above)
Comment (JiaChui) [reply zcommon/src/org/zkoss/video/AVideo.java]: ok

File: zcommon/src/org/zkoss/video/AVideo.java (hunk 5 above)
Comment (JiaChui) [reply zcommon/src/org/zkoss/video/AVideo.java]: ok

**Pull request merged**